maas/resource_maas_boot_source.go

```diff
@@ -1,12 +1,150 @@
 package maas
 
 import (
+	"context"
 	"fmt"
 
 	"github.com/canonical/gomaasclient/client"
 	"github.com/canonical/gomaasclient/entity"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
 )
 
+const defaultURL = "http://images.maas.io/ephemeral-v3/stable/"
+const defaultKeyring = "/snap/maas/current/usr/share/keyrings/ubuntu-cloudimage-keyring.gpg"
```

Comment:
this is a snap specific const. I think we should have a deb sepcific const too. In addition, if MAAS cannot provide via an API call whether it is a snap or deb installation, we may have to introduce a provider configuration. By doing so, users can tell the provider about the installation type and provider can be aware and decide things like this based on the value. The default value could be snap and the option would definitely be optional. We also have to add extra documentation to explain to users what is the purpose of this new provider config.